etcdserver/server.go

```diff
@@ -1646,7 +1646,11 @@ func (s *EtcdServer) PromoteMember(ctx context.Context, id uint64) ([]*membershi
 	// fails with ErrNotLeader, forward the request to leader node via HTTP. If promoteMember call fails with error
 	// other than ErrNotLeader, return the error.
 	resp, err := s.promoteMember(ctx, id)
+	if err == nil {
```

Comment:
If `err == nil`, `learnerPromoteFailed` will also increase because err != ErrNotLeader is also true. Maybe check `err == nil` inside `err != ErrNotLeader`?

Basically `err != ErrNotLeader` means the local member is leader and the promotion is either successful or failed (no need to forward the request). Once this condition is satisfied, we can collect metrics.

Reply:
I realized the code comment is confusing, I will update it.

Reply:
I use early return instead of nested if clause since it's more readable